File: test/DCG/TestData.hs
module DCG.TestData where

import DCG.Grammar
import Util.Container

noun = Term "Noun"
verb = Term "Verb"
det  = Term "Det"
prep = Term "Prep"

lexicon = asMap [ "fly"  ~> [noun, verb],
                  "like"   ~> [prep, verb],
                  "time"   ~> [noun],
                  "arrow"  ~> [noun],
                  "these"  ~> [det],
                  "planes" ~> [noun],
                  "an"     ~> [det] ]

utterance = ["these", "planes", "fly", "like", "an", "arrow"]

grammar = Grammar "S" ["S"  ==> ["NP", "VP"],
                       "VP" ==> ["Verb"],
                       "VP" ==> ["Verb", "NP"],
                       "VP" ==> ["VP", "PP"],
                       "NP" ==> ["Noun"],
                       "NP" ==> ["Det", "Noun"],
                       "NP" ==> ["NP", "PP"],
                       "PP" ==> ["Prep", "NP"]
                       ]

getResource:: FilePath -> (String -> IO()) -> IO()
getResource path f =
    do content <- readFile $ "./resources/" ++ path
       f content

File: src/test/scala/org/lolczak/dcg/parser/TestData.scala
package org.lolczak.dcg.parser

import org.lolczak.dcg.model.Grammar._
import org.lolczak.dcg.model._
import org.lolczak.dcg.parser.grammar.GrammarParser.{keyword => _}

import scala.Predef.{augmentString => _, wrapString => _, _}

object TestData {

  val lexicon = new SimpleLexicon(
    "fly" -> Set[Term]("Verb", "Noun"("Num" -> FConst("sg"))),
    "like" -> Set[Term]("Verb", "Prep"),
    "arrow" -> Set[Term]("Noun"("Num" -> FConst("sg"))),
    "these" -> Set[Term]("Det"("Num" -> FConst("pl"))),
    "this" -> Set[Term]("Det"("Num" -> FConst("sg"))),
    "planes" -> Set[Term]("Noun"("Num" -> FConst("pl"))),
    "an" -> Set[Term]("Det"("Num" -> FConst("sg")))
  )
  
  val nonterminals = Nonterminals("S",
    List(
      "S" ~>("NP", "VP"),
      "VP" ~> "Verb",
      "VP" ~>("Verb", "NP"),
      "VP" ~>("VP", "PP"),
      "ERule" ~> ("Empty", "Sth"),
      "Empty" ~> (),
      "NP"("Num" -> FVariable("n")) ~> "Noun"("Num" -> FVariable("n")),
      "NP"("Num" -> FVariable("n")) ~>("Det"("Num" -> FVariable("n")), "Noun"("Num" -> FVariable("n"))),
      "NP"("Num" -> FVariable("n")) ~>("NP"("Num" -> FVariable("n")), "PP"),
      "PP" ~>("Prep", "NP")
    )
  )

  val grammar = Grammar(nonterminals, lexicon, List.empty)

  val utterance = "these planes fly like an arrow"

  val incorrectUtterance = "this planes fly like an arrow"

}
